### ISSUE TITLE
Messages are not marked as read

### ISSUE THREAD

**Issue by cjmconie** [GetStream/stream-chat-swift#933]
## What did you do?

View a channel's unread messages by entering the message list, then navigating back to the channel list.

## What did you expect to happen?

The unread message indicator (right hand side of the `ChatChannelListItemView`) to decrease to zero.

## What happened instead?

The channel's unread count remains unchanged. In fact, each channel's unread count only seems to increase. 
The app's badge is also affected (since we have push notifications enabled).

## GetStream Environment
**GetStream Chat version:** 3.2.0-beta4
**GetStream Chat frameworks:** StreamChat, StreamChatUI, StreamChatClient, StreamChatCore
**iOS version:** 13.0
**Swift version:** 5.3.2
**Xcode version:** 12.4
**Device:** Any

## Additional context

I see this behaviour on both the Demo app and when integrating StreamChatUI into our own app.

Possibly related to [Issue 897](https://github.com/GetStream/stream-chat-swift/issues/897)

**Comment by cjmconie** [comment]
Just an update, subclassing `ChatChannelVC` and calling `markRead()` in `viewDidAppear`

```swift
class CustomChatChannelVC: ChatChannelVC {
  
  override func viewDidAppear(_ animated: Bool) {
    guard let channel = channelController.channel else { return }
    
    channelController.markRead()
    
  }
```

results in the unread count being set correctly, however the UI components only update after a terminate/launch of the app.

Possibly related, I noticed that both unread publishers (`currentUserController().observableObject.$unreadCount` and `currentUserController().unreadCountPublisher`) aren't called after `markRead()` is called.

**Comment by VojtaStavik** [comment]
Hi @cjmconie, thank you for providing more info. We will check this issue shortly.

**Comment by cjmconie** [comment]
@VojtaStavik I see in [3.2.0-beta.6](https://github.com/GetStream/stream-chat-swift/releases/tag/3.2.0-beta.6) there has been some progress on this issue, thanks! 

The channel's unread count correctly decreases when entering presenting a `ChatChannelVC`.

Issues which we're still facing:
1. Unread counts do not decrease when receiving a message while a `ChatChannelVC` is on screen.
2. Unread publishers are not called.

**Comment by b-onc** [comment]
Hello @cjmconie ,

Issue 1 is in our backlog, we'll introduce it soon. Issue 2 should've been resolved, are you still seeing it?

**Comment by b-onc** [comment]
Hello @cjmconie 

This issue is fixed by #1147, please re-open if you still see this.
